### ISSUE TITLE
Problem with last data output from throttle

### ISSUE THREAD

**Issue by Funncy** [ReactiveX/rxdart#417]
```Dart 
  Stream.fromIterable([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
      .interval(Duration(milliseconds: 500))
      .throttle((data) {
    print('throttle data =$data');
    return TimerStream(true, Duration(seconds: 1));
  }, trailing: false).listen(print);
```

This is my example code.

As far as I know, the above code in the Throttle action should not produce the last data. However, the last data is output.

The particular thing is that if the trailing value is given true, the last data will not be output normally.

---

When I read backpress.dart, it is a structure that clears Queue from singleWindow onDone. 
```Dart
final singleWindow = (S event) => buildStream(event).take(1).listen(
                null,
                onError: controller.addError,
                onDone: resolveWindowEnd,
                cancelOnError: cancelOnError);
```

```Dart
final onDone = () {
              // treat the final event as a Window that opens
              // and immediately closes again
              if (queue.isNotEmpty) resolveWindowStart(queue.last);

              resolveWindowEnd(true);

              queue.clear();
              controller.close();
            };
```
When the main stream ends normally, the OnDone in the main stream is called instead of the OnDone in the single window, so the last queue is not cleaned up and the resetWindow function is executed once more.

I added Queue.clear to the top, and it worked.

I am studying RxDart now. Please help me.

**Comment by frankpepermans** [comment]
good catch! see: https://github.com/ReactiveX/rxdart/pull/418

**Comment by brianegan** [comment]
Fixed by #418
